### PR TITLE
SDIT-1568 Various alerts mapping endpoints 

### PR DESCRIPTION
**Pull request by andymarke**
* POST /mapping/alerts/batch  - require to created mappings after a prisoner merge
* GET /mapping/alerts/{offenderNo}/all - utility for diagnostics
* GET /mapping/alerts - utility for diagnostics

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/alerts/AlertMappingService.kt [hunk 1]
```diff
@@ -92,6 +97,22 @@ class AlertMappingService(
...
   }
 
+  suspend fun getMappings(pageRequest: Pageable): Page<AlertMappingDto> = coroutineScope {
+    val mappings = repository.findAllBy(
+      pageRequest = pageRequest,
+    )
+
+    val count = async {
+      repository.count()
+    }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/alerts/AlertMappingService.kt:107]: Why just 1 async call? Not sure how this helps?

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/alerts/AlertMappingService.kt:107]: Well the `mappings.toList()`  is the 2nd hidden one (I think!)

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/alerts/AlertMappingService.kt:107]: But actually, yes I think you right, I will do another PR to fix

**Review by lbennett-moj — APPROVED** [review]
lgtm

**Pull request merged**